sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -655,9 +660,13 @@ def process(
             t=beam.DoFn.TimestampParam):
           yield test_stream.WindowedValueHolder(WindowedValue(e, t, [w], p))
 
+      extended_target = pcoll
+      if self._is_streaming_pcollection(pcoll):
```

Comment:
We should *always* be writing windowing information to the cache, as even in the global window there may be other important information such as timestamps that we need to preserve. This simplifies the code below as well. 

We can conditionally decide to display this (i.e. have the default value of `include_window_info` depend on whether the windowing is default, which the user can manually specify one way or the other if that's not what they want).

Reply:
This is why I was interested in figuring out whether pcollection is not streaming, so that windowing wouldn't matter. I don't fully understand the code, but the comment to Reify suggests that wrapping is also needed outside the visualization ('When it detects one, it puts the element into the correct window then emits the value to downstream transforms').

It seems nice to have this just depend on the flag, but are we sure it would work pcollections with non-trivial windowing? I also can imagine a case where ib.collect is called with `include_window_info=False` at first and then with `include_window_info=True`, which probably would result in an error.

Reply:
Windowing can be present (and matter) in batch pipelines. Conversely, streaming pipelines can use the Global windows. The correct thing to do is always store this information in the cache, and conditionally strip it out in `ib.collect` if it's "not interesting."